logic/src/commonMain/kotlin/com/wire/kalium/logic/data/publicuser/PublicUserRepository.kt

```diff
@@ -0,0 +1,58 @@
+package com.wire.kalium.logic.data.publicuser
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.data.publicuser.model.PublicUserSearchResult
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.mapLeft
+import com.wire.kalium.logic.wrapApiRequest
+import com.wire.kalium.network.api.contact.search.ContactSearchApi
+import com.wire.kalium.network.api.contact.search.ContactSearchRequest
+import com.wire.kalium.network.api.user.details.ListUserRequest
+import com.wire.kalium.network.api.user.details.UserDetailsApi
+import com.wire.kalium.network.api.user.details.qualifiedIds
+import com.wire.kalium.network.utils.flatMap
+import com.wire.kalium.network.utils.mapSuccess
+
+
+interface PublicUserRepository {
+    suspend fun searchPublicContact(
+        searchQuery: String,
+        domain: String,
+        resultSize: Int? = null
+    ): Either<CoreFailure, PublicUserSearchResult>
+}
+
+class PublicUserRepositoryImpl(
+    private val contactSearchApi: ContactSearchApi,
+    private val userApi: UserDetailsApi,
+    private val publicUserMapper: PublicUserMapper,
+) : PublicUserRepository {
+
+    override suspend fun searchPublicContact(
+        searchQuery: String,
+        domain: String,
+        resultSize: Int?
+    ): Either<NetworkFailure, PublicUserSearchResult> {
+        return wrapApiRequest {
+            contactSearchApi.search(
+                ContactSearchRequest(
+                    searchQuery = searchQuery,
+                    domain = domain,
+                    resultSize = resultSize
+                )
+            ).flatMap { contactResponse ->
+                val contactResultValue = contactResponse.value
+
+                userApi.getMultipleUsers(ListUserRequest.qualifiedIds(contactResultValue.documents.map { it.qualifiedID }))
+                    .mapSuccess { userDetailsResponses ->
+                        PublicUserSearchResult(
+                            totalFound = contactResultValue.found,
+                            publicUsers = publicUserMapper.fromUserDetailResponses(userDetailsResponses)
+                        )
+                    }
+            }
+        }.mapLeft { NetworkFailure.ServerMiscommunication(it.kaliumException) }
```

Comment:
`mapSuccess` is network internal function (for iOS compatibility) there is a fix in #258 the code can be rewritten as
```kotlin
wrapApiRequest{
    userApi.getMultipleUsers(ListUserRequest.qualifiedIds(contactResultValue.documents.map { it.qualifiedID }))
}.map {
    userDetailsResponses ->
    PublicUserSearchResult(
        totalFound = contactResultValue.found, publicUsers = publicUserMapper.fromUserDetailResponses(userDetailsResponses)
    )
}
```